### PR TITLE
nodejs-week3-meal-sharing

### PR DESCRIPTION
**Pull request by Nelia11**

### CODE REVIEW

File: src/backend/api/controllers/reviewControllers.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            "title",
+            "description",
+            "meal_id",
+            "stars",
+            "created_date"
+        );
+
+        reviews.length === 0
+        ? res.status(404).json({"error": "Reviews not found"})
+        : res.status(400).json(reviews);
```
Comment (raihanmehran) [inline src/backend/api/controllers/reviewControllers.js]: the status codes need to be updated. And I would suggest to just return the reviews object whether it contains the reviews or not.

File: src/backend/api/controllers/reviewControllers.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        .from("review")
+        .select(
+            "review.title" , 
+            "review.description", 
+            "stars", 
+            "meal.title AS meal", 
+            "location",
+            "review.created_date"
+            )
+        .leftJoin("meal", {"meal.id": "meal_id"})
```
Comment (raihanmehran) [inline src/backend/api/controllers/reviewControllers.js:46]: Well, In my perception I `meal_id` should be `review.meal_id`. I'm not so sure, and yes, according to the syntax it should `review.meal_id`.

File: src/backend/api/controllers/reviewControllers.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        ? res.status(404).json({"error": "Reviews not found"})
+        : res.status(400).json(reviews);
+    } catch (error) {
+        res.status(500).json(error);
+    }
+};
+
+const getMealReviews = async (req, res) => {
+    try {
+        const id = req.params.meal_id;
```
Comment (raihanmehran) [inline src/backend/api/controllers/reviewControllers.js:26]: Validating the `id` could be a good option to go for :)

File: src/backend/api/controllers/reviewControllers.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+            "title",
+            "description",
+            "meal_id",
+            "stars",
+            "created_date"
+        ];
+
+        const missingMandatoryField = requiredFields.some((field) => !review[field]);
+
+        if (missingMandatoryField) {
```
Comment (raihanmehran) [inline src/backend/api/controllers/reviewControllers.js:79]: appreciated 💯

**Review by raihanmehran — COMMENTED** [review]
Nice Solution 👍

**Comment by Nelia11** [comment]
@raihanmehran  Thank you a lot for the review and your help during the NodeJS module!

**Comment by raihanmehran** [comment]
> @raihanmehran Thank you a lot for the review and your help during the NodeJS module!

Always welcome 🙏, if you have any further questions, feel free to reach out.

**Pull request merged**